parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -256,4 +260,64 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return Collections.unmodifiableMap(setMultiMap);
   }
 
+  //TODO maintain unmaterializableRecordCounter properly
+  public boolean nextBatch(ObjectColumnVector<T> vector) throws IOException {
+    boolean recordFound = false;
+
+    while (!recordFound) {
+      // no more records left
+      if (current >= total) { return false; }
+
+      try {
+        checkRead();
+        ((VectorizedRecordReader) recordReader).readVector(vector, current, totalCountLoadedSoFar);
+        current += vector.size();
+
+        //TODO filtering records while batch reading
+        //TODO how to handle this case?
+//        if (currentValue == null) {
+//          // only happens with FilteredRecordReader at end of block
+//          current = totalCountLoadedSoFar;
+//          if (DEBUG) LOG.debug("filtered record reader reached end of block");
+//          continue;
+//        }
+
+        recordFound = true;
+      } catch (RuntimeException e) {
+        throw new ParquetDecodingException(format("Can not read value at %d in block %d in file %s", current, currentBlock, file), e);
+      }
+    }
+    return true;
+  }
+
+  //TODO maintain unmaterializableRecordCounter properly
+  public boolean nextBatch(ColumnVector[] vectors, MessageType[] columns) throws IOException {
+    boolean recordFound = false;
+
+    while (!recordFound) {
+      // no more records left
+      if (current >= total) { return false; }
+
+      try {
+        checkRead();
+        ((VectorizedRecordReader) recordReader).readVectors(vectors, columns, current, totalCountLoadedSoFar);
```

Comment:
Doesn't this assume that there are no repeated values? I don't see this checking repetition level anywhere.